qiskit/pulse/commands/parametric_pulses.py

```diff
@@ -0,0 +1,337 @@
+
+        Raises:
+            PulseError: If the parameters passed are not valid.
+        """
+        pass
+
+    def get_params(self) -> Dict[str, Any]:
+        """Return a dictionary containing the parameters specific to this parametric pulse."""
+        return {attr[1:]: getattr(self, attr)
+                for attr in self.__dict__
+                if attr.startswith('_') and attr != '_name'}
+
+    def to_instruction(self, channel: PulseChannel,
+                       name: Optional[str] = None) -> 'ParametricInstruction':
+        # pylint: disable=arguments-differ
+        return ParametricInstruction(self, channel, name=name)
+
+    def __repr__(self):
+        return '{}(name={}, params={})'.format(self.__class__.__name__,
+                                               self.name,
+                                               self.get_params())
+
+
+class Gaussian(ParametricPulse):
+    """
+    A truncated pulse envelope shaped according to the Gaussian function whose mean is centered at
+    the center of the pulse (duration / 2):
+
+        f(x) = amp * exp( -(1/2) * (x - duration/2)^2 / sigma^2) )  ,  0 <= x < duration
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float):
+        """Initialize the gaussian command.
+
+        Args:
+            duration: Pulse length in terms of the the sampling period `dt`.
+            amp: The amplitude of the Gaussian envelope.
+            sigma: A measure of how wide or narrow the Gaussian peak is; described mathematically
+                   in the class docstring.
+        """
+        self._amp = complex(amp)
+        self._sigma = sigma
+        super().__init__(duration=duration)
+        self.validate_params()
+
+    @property
+    def amp(self):
+        return self._amp
+
+    @property
+    def sigma(self):
+        return self._sigma
+
+    def get_sample_pulse(self) -> SamplePulse:
+        return gaussian(duration=self.duration, amp=self.amp,
+                        sigma=self.sigma, zero_ends=False)
+
+    def validate_params(self) -> None:
+        if abs(self.amp) > 1.:
+            raise PulseError("The amplitude norm must be <= 1, "
+                             "found: {}".format(abs(self.amp)))
+        if self.sigma <= 0:
+            raise PulseError("Sigma must be greater than 0.")
+
+
+class GaussianSquare(ParametricPulse):
+    """
+    A square pulse with a Gaussian shaped risefall on either side:
+
+        risefall = (duration - width) / 2
+
+    0 <= x < risefall
+        f(x) = amp * exp( -(1/2) * (x - risefall/2)^2 / sigma^2) )
+
+    risefall <= x < risefall + width
+        f(x) = amp
+
+    risefall + width <= x < duration
+        f(x) = amp * exp( -(1/2) * (x - (risefall + width)/2)^2 / sigma^2) )
+    """
+
+    def __init__(self,
+                 duration: int,
+                 amp: complex,
+                 sigma: float,
+                 width: float):
```

Comment:
Could you please remind me why we settled on width instead of risefall? I think this could be conflated with `duration`.